Assets/Scripts/HandController.cs

```diff
@@ -2,56 +2,66 @@
 using System.Collections.Generic;
 using UnityEngine;
 
-public class HandController : MonoBehaviour
-{
-    public List<GameObject> cards;
-    public List<GameObject> selectedCards;
+public class HandController : MonoBehaviour {
+    public List<CardController> selectedCards;
     public int maxCards;
     public GameObject cardPrefab;
 
     public CastButtonController castButton;
 
-    void Start()
-    {
-        GenerateCards();
+    void Start () {
+        GenerateCards ();
     }
 
-    public void SelectCard(GameObject card) {
-        if (!selectedCards.Contains(card)) {
-            selectedCards.Add(card);
+    public void SelectCard (CardController cardController) {
+        selectedCards.Add (cardController);
+        // Using List.Count to retain the index+1 value for player readability
+        bool canCast = castButton.OnSelectionChanged (selectedCards.Count);
+        if (canCast) {
+            UpdateAllSelectedCards (canCast);
         } else {
-            selectedCards.Remove(card);
+            cardController.UpdateSelectedCardIndex (selectedCards.Count, castButton.OnSelectionChanged (selectedCards.Count));
         }
+    }
+
+    public void DeselectCard (CardController cardController) {
+        selectedCards.Remove (cardController);
+        cardController.UpdateSelectedCardIndex (0);
+        UpdateAllSelectedCards (castButton.OnSelectionChanged (selectedCards.Count));
+    }
 
-        castButton.OnSelectionChanged(selectedCards.Count);
+    int RetrieveIndex (CardController cardController) {
+        return selectedCards.IndexOf (cardController) + 1;
     }
 
-    void GenerateCards() {
+    void GenerateCards () {
```

Comment:
Oh do we have a thing about spaces before param/arg declaration?

Reply:
This is just how auto-formatting with option + shift + f  does it 🤷🏻‍♂️

Reply:
I don't know if I like it lol